codex-rs/app-server-protocol/src/protocol/common.rs

```diff
@@ -366,16 +506,67 @@ pub struct FuzzyFileSearchResponse {
     pub files: Vec<FuzzyFileSearchResult>,
 }
 
-/// Notification sent from the server to the client.
-#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema, TS, Display)]
-#[serde(tag = "method", content = "params", rename_all = "camelCase")]
-#[strum(serialize_all = "camelCase")]
-pub enum ServerNotification {
+server_notification_definitions! {
     /// NEW NOTIFICATIONS
+    #[serde(rename = "thread/started")]
+    #[ts(rename = "thread/started")]
+    #[strum(serialize = "thread/started")]
+    ThreadStarted(v2::ThreadStartedNotification),
+
+    #[serde(rename = "thread/error")]
+    #[ts(rename = "thread/error")]
+    #[strum(serialize = "thread/error")]
+    ThreadError(v2::ThreadErrorNotification),
+
+    #[serde(rename = "turn/started")]
+    #[ts(rename = "turn/started")]
+    #[strum(serialize = "turn/started")]
+    TurnStarted(v2::TurnStartedNotification),
+
+    #[serde(rename = "turn/completed")]
+    #[ts(rename = "turn/completed")]
+    #[strum(serialize = "turn/completed")]
+    TurnCompleted(v2::TurnCompletedNotification),
+
+    #[serde(rename = "turn/failed")]
+    #[ts(rename = "turn/failed")]
+    #[strum(serialize = "turn/failed")]
+    TurnFailed(v2::TurnFailedNotification),
+
+    #[serde(rename = "item/started")]
+    #[ts(rename = "item/started")]
+    #[strum(serialize = "item/started")]
+    ItemStarted(v2::ItemStartedNotification),
+
+    #[serde(rename = "item/updated")]
+    #[ts(rename = "item/updated")]
+    #[strum(serialize = "item/updated")]
+    ItemUpdated(v2::ItemUpdatedNotification),
+
+    #[serde(rename = "item/completed")]
+    #[ts(rename = "item/completed")]
+    #[strum(serialize = "item/completed")]
+    ItemCompleted(v2::ItemCompletedNotification),
+
+    #[serde(rename = "item/agentMessage/delta")]
+    #[ts(rename = "item/agentMessage/delta")]
+    #[strum(serialize = "item/agentMessage/delta")]
+    AgentMessageDelta(v2::AgentMessageDeltaNotification),
```

Comment:
Should this type name also start with `Item`?

Is our naming convention that it is the "last two" parts of the path?